test/enssec.js

```diff
@@ -275,6 +283,78 @@ contract('DNSSEC', function(accounts) {
     await verifyFailedSubmission(instance, test_rrsets[0][0], "0x" + test_rrsets[0][1], "0x" + sig.slice(0, sig.length - 2) + "FF");
   });
 
+  // Test delete RRsec
+  async function checkPresence(instance, type, name){
+    var result = (await instance.rrset.call(1, type, dns.hexEncodeName(name)))[2];
+    return result != '0x';
+  }
+
+  async function submitEntry(instance, type, name, option){
+    var rrs = {name: name, type: type, klass: 1, ttl: 3600};
+    Object.assign(rrs, option)
+    var keys = {
+      typeCovered: type,
+      algorithm: 253,
+      labels: 1,
+      originalTTL: 3600,
+      expiration: 0xFFFFFFFF,
+      inception: 1,
+      keytag: 5647,
+      signerName: ".",
+      rrs: [rrs],
+    };
+    var [inception, _, rrs] = await instance.rrset.call(1, type, dns.hexEncodeName(name));
+    if(rrs != '0x'){
+      keys.inception=inception + 1;
+    };
+    trx = await verifySubmission(instance, name, dns.hexEncodeSignedSet(keys), "0x");
+    [_, _, rrs] = await instance.rrset.call(1, type, dns.hexEncodeName(name));
+    assert.notEqual(rrs, '0x');  
+  }
+
+  it('rejects if NSEC record is not found', async function(){
+    var instance = await dnssec.deployed();
+    await submitEntry(instance, dns.TYPE_TXT, 'b.', {text: ["foo"]});
+    await instance.deleteRRSet(1, dns.hexEncodeName('a.'), dns.TYPE_TXT, dns.hexEncodeName('b.'));
+    assert.equal((await checkPresence(instance, dns.TYPE_TXT, 'b.')), true);
+  })
+
+  it('rejects if NSEC record does not match deleting record type', async function(){
+    var instance = await dnssec.deployed();
+    await submitEntry(instance, dns.TYPE_TXT, 'b.', {text: ["foo"]});
+    // proving record is not NSEC
+    var option = {klass: dns.CLASS_INET, ttl: 3600, flags: 0x0101, protocol: 3, algorithm: 253, pubkey: new Buffer("1111", "HEX")}
+    await submitEntry(instance, dns.TYPE_DNSKEY, 'a.',  option);
+    await instance.deleteRRSet(1, dns.hexEncodeName('a.'), dns.TYPE_TXT, dns.hexEncodeName('b.'));
+    assert.equal((await checkPresence(instance, dns.TYPE_TXT, 'b.')), true);
+  })
+
+  it('rejects if next record does not come before the deleting name', async function(){
```

Comment:
You should test the inverse too - that you can't delete a record with an NSEC record that starts after the name.